### PR TITLE
adding failing test for returning cloned modified object from post hook

### PR DESCRIPTION
**Pull request by brettmc**
Cloning, then modifying and returning the return value from a post hook can cause either a segfault or a reported memory leak (it seems random, see comment in test).
The use-case for this is changing an immutable object (such as a PSR-7 message) from a post hook.

### CODE REVIEW

File: ext/tests/post_hook_returns_cloned_modified_object.phpt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+});
+
+function getFoo(): Foo {
+    return new Foo('a'); //providing a constructor arg or not sometimes changes error to a reported memory leak
+}
+
+var_dump(getFoo());
+?>
+--EXPECT--
+object(Foo)#2 (1) {
```
Comment (agoallikmaa) [inline ext/tests/post_hook_returns_cloned_modified_object.phpt:37]: After the fix, this apparently becomes `#3` in my testing. Not sure how stable this is though, so perhaps best to use `EXPECTF` and `#%d` here.

File: ext/tests/post_hook_returns_cloned_modified_object.phpt
Comment (agoallikmaa) [inline ext/tests/post_hook_returns_cloned_modified_object.phpt:1]: Should add the file to `package.xml`.

**Review by agoallikmaa — CHANGES_REQUESTED** [review]
Implemented the fix in https://github.com/open-telemetry/opentelemetry-php-instrumentation/pull/136 and checked that this test passes then (after the minor change to expected output I mentioned).

File: ext/tests/post_hook_returns_cloned_modified_object.phpt (hunk 1 above)
Comment (brettmc) [reply ext/tests/post_hook_returns_cloned_modified_object.phpt]: agreed, updated.

**Comment by brettmc** [comment]
test is passing after applying the fix from #136, just need an approver please @bobstrecansky

**Review by bobstrecansky — APPROVED** [review]

**Pull request merged**